File: VirtualTeacherGenAIDemo.Server/Models/Request/ChatHistoryRequest.cs
using OpenAI.Chat;
using System.ComponentModel.DataAnnotations;
using VirtualTeacherGenAIDemo.Server.Models.Storage;

namespace VirtualTeacherGenAIDemo.Server.Models.Request
{
    public class ChatHistoryRequest
    {
        public string UserId { get; set; } = string.Empty;
        public SessionItem? Session { get; set; }
        public List<ChatMessage> Messages { get; set; } = new();
    }

    public class ChatMessage
    {
        public AuthorRole Role { get; set; }
        public string Content { get; set; } = string.Empty;
        public string Id { get; set; } = string.Empty;
    }

    public enum AuthorRole
    {
        User,
        Assistant,
        System
    }
}

File: virtualteachergenaidemo.client/src/models/ChatHistoryRequest.ts
import { SessionItem } from '../models/SessionItem';

export enum AuthorRole {
    User,
    Assistant,
    System
}

export class ChatHistoryRequest {
    messages: ChatMessage[];
    userId: string;
    session: SessionItem;

    constructor(userId: string, session: SessionItem, messages: ChatMessage[]) {
        this.userId = userId;
        this.session = session;
        this.messages = messages;
    }
}

export class ChatMessage {
    role: AuthorRole;
    content: string;
    id: string;

    constructor(id: string, role: AuthorRole, content: string) {
        this.id = id;
        this.role = role;
        this.content = content;
    }
}
